api-report/tree.api.md

```diff
@@ -340,6 +346,29 @@ export interface ITreeCursor<TResult = TreeNavigationResult> {
     readonly value: Value;
 }
 
+// @public
+export interface ITreeCursorNew {
+    readonly currentChunkLength: number;
+    readonly currentChunkStart: number;
+    readonly currentIndexInField: number;
+    enterChildNode(index: number): void;
+    enterField(key: FieldKey): void;
+    getCurrentFieldKey(): FieldKey;
+    // (undocumented)
+    getCurrentFieldLength(): number;
+    // (undocumented)
+    getPath(): UpPath | undefined;
+    readonly mode: CursorLocationType;
+    nextField(skipPending: boolean): boolean;
+    // (undocumented)
+    readonly pending: boolean;
+    seek(offset: number): boolean;
+    readonly type: TreeType;
+    upToField(): void;
+    upToNode(): void;
+    readonly value: Value;
+}
```

Comment:
I made these changes. @CraigMacomber, could you answer the part about get methods not being properties?

Reply:
Decision was pretty arbitrary. Generally using getters instead of properties implies there is cost to reading those values and caching them in a local instead of rereading them may be a good idea as Daniel mentioned, but I don't really have much reason to think any of these will be particularly expensive. As this is an interface, and some implementation might have them be somewhat expensive, I use getters for some of them. Once we have some more real (compressed) implementations we can reevaluate it.